packages/firebase_messaging/firebase_messaging_platform_interface/lib/src/platform_interface/platform_interface_messaging.dart

```diff
@@ -0,0 +1,309 @@
+  /// background or terminated.
+  ///
+  /// This provided handler must be a top-level function and cannot be
+  /// anonymous otherwise an [ArgumentError] will be thrown.
+  static BackgroundMessageHandler get onBackgroundMessage {
+    return _onBackgroundMessageHandler;
+  }
+
+  /// Allows the background message handler to be created and calls the
+  /// instance delegate [registerBackgroundMessageHandler] to perform any
+  /// platform specific registration logic.
+  static set onBackgroundMessage(BackgroundMessageHandler handler) {
+    _onBackgroundMessageHandler = handler;
+    instance.registerBackgroundMessageHandler(handler);
+  }
+
+  /// Enables delegates to create new instances of themselves if a none default
+  /// [FirebaseApp] instance is required by the user.
+  @protected
+  FirebaseMessagingPlatform delegateFor({FirebaseApp app}) {
+    throw UnimplementedError('delegateFor() is not implemented');
+  }
+
+  /// Sets any initial values on the instance.
+  ///
+  /// Platforms with Method Channels can provide constant values to be available
+  /// before the instance has initialized to prevent any unnecessary async
+  /// calls.
+  @protected
+  FirebaseMessagingPlatform setInitialValues({
+    bool isAutoInitEnabled,
+  }) {
+    throw UnimplementedError('setInitialValues() is not implemented');
+  }
+
+  /// Returns whether messaging auto initialization is enabled or disabled for the device.
+  bool get isAutoInitEnabled {
+    throw UnimplementedError('isAutoInitEnabled is not implemented');
+  }
+
+  /// If the application has been opened from a terminated state via a [RemoteMessage]
+  /// (containing a [Notification]), it will be returned, otherwise it will be `null`.
+  ///
+  /// Once the [Notification] has been consumed, it will be removed and further
+  /// calls to [getInitialMessage] will be `null`.
+  ///
+  /// This should be used to determine whether specific notification interaction
+  /// should open the app with a specific purpose (e.g. opening a chat message,
+  /// specific screen etc).
+  Future<RemoteMessage> getInitialMessage() {
+    throw UnimplementedError('getInitialMessage() is not implemented');
+  }
+
+  /// Allows delegates to create a background message handler implementation.
+  ///
+  /// For example, on native platforms this could be to setup an isolate, whereas
+  /// on web a service worker can be registered.
+  void registerBackgroundMessageHandler(BackgroundMessageHandler handler) {
+    throw UnimplementedError(
+        'registerBackgroundMessageHandler() is not implemented');
+  }
+
+  /// Removes access to an FCM token previously authorized with optional [senderId].
+  ///
+  /// Messages sent by the server to this token will fail.
+  Future<void> deleteToken({
+    String senderId,
+  }) {
+    throw UnimplementedError('deleteToken() is not implemented');
+  }
+
+  /// On iOS, it is possible to get the users APNs token. This may be required
```

Comment:
```suggestion
  /// On iOS & MacOS, it is possible to get the users APNs token. This may be required
```
same again re: geAPNSToken() if it's possible? :)